src/lib/core/private.h

```diff
@@ -0,0 +1,126 @@
+/* Private header for tzdb code.  */
```

Comment:
Putting `private.h` to `src/lib/core` directory seemed wrong idea, and considering the fact that we would need to add `strptime()` implementation, in addition to the `strftime()`, and they both have to be extended with `"%f"` flag support, it was decided to move `private.h`/`strftime.c` to their special 3rd party directory, like `third_party/tzcode/`, the way like Android does similar code reuse https://android.googlesource.com/platform/bionic/+/master/libc/tzcode/ (please see that `strftime.c` there is from Olson, while `strptime.c` is from NetBSD - we will proceed similarly).

Here is the patch to the 2nd commit which rename files, and create 3rd party library:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8037c30a7..1bbfba25a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -579,6 +579,14 @@ include(BuildCDT)
 libccdt_build()
 add_dependencies(build_bundled_libs cdt)
 
+#
+# portion of Olson tzcode
+#
+
+include(BuildTZ)
+libtzcode_build()
+add_dependencies(build_bundled_libs tzcode)
+
 #
 # Third-Party misc
 #
diff --git a/cmake/BuildTZ.cmake b/cmake/BuildTZ.cmake
new file mode 100644
index 000000000..dd4733300
--- /dev/null
+++ b/cmake/BuildTZ.cmake
@@ -0,0 +1,5 @@
+macro(libtzcode_build)
+    set(LIBTZ_LIBRARIES tzcode)
+
+    add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/tzcode)
+endmacro()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4473ff1da..b337f3859 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -196,7 +196,8 @@ target_link_libraries(server core coll http_parser bit uri uuid swim swim_udp
 # library here.
 set (reexport_libraries server core misc bitset csv swim swim_udp swim_ev
      shutdown ${LUAJIT_LIBRARIES} ${MSGPUCK_LIBRARIES} ${ICU_LIBRARIES}
-     ${CURL_LIBRARIES} ${XXHASH_LIBRARIES} ${LIBCDT_LIBRARIES})
+     ${CURL_LIBRARIES} ${XXHASH_LIBRARIES} ${LIBCDT_LIBRARIES}
+     ${LIBTZ_LIBRARIES})
 
 set (common_libraries
     ${reexport_libraries}
diff --git a/src/lib/core/CMakeLists.txt b/src/lib/core/CMakeLists.txt
index 21508497f..8bc776b82 100644
--- a/src/lib/core/CMakeLists.txt
+++ b/src/lib/core/CMakeLists.txt
@@ -31,7 +31,6 @@ set(core_sources
     mp_decimal.c
     cord_buf.c
     datetime.c
-    strftime.c
 )
 
 if (TARGET_OS_NETBSD)
diff --git a/third_party/tzcode/CMakeLists.txt b/third_party/tzcode/CMakeLists.txt
new file mode 100644
index 000000000..f76721231
--- /dev/null
+++ b/third_party/tzcode/CMakeLists.txt
@@ -0,0 +1,5 @@
+set(tzcode_sources strftime.c)
+
+add_library(tzcode STATIC ${tzcode_sources})
+
+target_link_libraries(tzcode core)
diff --git a/src/lib/core/private.h b/third_party/tzcode/private.h
similarity index 100%
rename from src/lib/core/private.h
rename to third_party/tzcode/private.h
diff --git a/src/lib/core/strftime.c b/third_party/tzcode/strftime.c
similarity index 100%
rename from src/lib/core/strftime.c
rename to third_party/tzcode/strftime.c
```